toolkit/tools/pkg/imagecustomizerlib/partitionutils.go

```diff
@@ -318,6 +321,22 @@ func findSourcePartitionHelper(source string,
 	return mountIdType, partition, partitionIndex, nil
 }
 
+func findExtendedSourcePartitionHelper(source string, partitions []diskutils.PartitionInfo,
+	buildDir string,
+) (imagecustomizerapi.ExtendedMountIdentifierType, diskutils.PartitionInfo, int, error) {
+	mountIdType, mountId, err := parseExtendedSourcePartition(source)
+	if err != nil {
+		return imagecustomizerapi.ExtendedMountIdentifierTypeDefault, diskutils.PartitionInfo{}, 0, err
+	}
+
+	partition, partitionIndex, err := findExtendedPartition(mountIdType, mountId, partitions, buildDir)
+	if err != nil {
+		return imagecustomizerapi.ExtendedMountIdentifierTypeDefault, diskutils.PartitionInfo{}, 0, err
+	}
+
+	return mountIdType, partition, partitionIndex, nil
+}
+
 func findPartition(mountIdType imagecustomizerapi.MountIdentifierType, mountId string,
```

Comment:
Can `findPartition` be implemented using `findExtendedPartition` to avoid duplicate code?